tests/runner.py

```diff
@@ -1354,30 +1354,41 @@ class BrowserCore(RunnerCore):
   def __init__(self, *args, **kwargs):
     super(BrowserCore, self).__init__(*args, **kwargs)
 
+  @staticmethod
+  def browser_open(url):
+    if not EMTEST_BROWSER:
+      print("Using default system browser")
+      webbrowser.open_new(url)
+      return
+
+    browser_args = shlex.split(EMTEST_BROWSER)
+    # If the given browser is a string with no separators, assume it's one
+    # of the types that the webbrowser module accepts like `safari`,
+    # `firefox`, or `chrome`. See https://docs.python.org/2/library/webbrowser.html
+    # for the full list
+    if len(browser_args) == 1 and os.path.sep not in browser_args[0]:
+      print("Using Emscripten browser: " + browser_args[0])
+      webbrowser.get(browser_args[0]).open_new(url)
+      return
+    # Else assume the given browser is a specific program with additional
+    # parameters and delegate to that
+    print("Using Emscripten browser: " + str(browser_args))
+    subprocess.Popen(browser_args + [url])
+
   @classmethod
   def setUpClass(cls):
     super(BrowserCore, cls).setUpClass()
     cls.also_asmjs = int(os.getenv('EMTEST_BROWSER_ALSO_ASMJS', '0')) == 1
     cls.port = int(os.getenv('EMTEST_BROWSER_PORT', '8888'))
     if not has_browser():
       return
-    if not EMTEST_BROWSER:
-      print("Using default system browser")
-    else:
-      cmd = shlex.split(EMTEST_BROWSER)
-
-      def run_in_other_browser(url):
-        subprocess.Popen(cmd + [url])
-
-      webbrowser.open_new = run_in_other_browser
-      print("Using Emscripten browser: " + str(cmd))
     cls.browser_timeout = 60
     cls.harness_in_queue = multiprocessing.Queue()
     cls.harness_out_queue = multiprocessing.Queue()
     cls.harness_server = multiprocessing.Process(target=harness_server_func, args=(cls.harness_in_queue, cls.harness_out_queue, cls.port))
     cls.harness_server.start()
     print('[Browser harness server on process %d]' % cls.harness_server.pid)
-    webbrowser.open_new('http://localhost:%s/run_harness' % cls.port)
+    cls.browser_open('http://localhost:%s/run_harness' % cls.port)
```

Comment:
I don't don't if it matters but this does mean that `[Browser harness server on process %d]` is not longer the last thing that is printed by `setUpClass`.     Otherwise seems reasonable.   I'll leave the final lg to @kripken